### ISSUE TITLE
Got disconnected instantly when using BungeeCord/Velocity

### ISSUE THREAD

**Issue by AchiraE** [ShetiPhian/Localizations-and-Issues#335]
**Describe the bug**

Players got disconnected immediately if the connection is through BungeeCord/Velocity. Direct connection to the server works fine.

Logs from BungeeCord:
```
[00:10:29 WARN]: [/********:13430|djytw] <-> DownstreamBridge <-> [lobby] - could not decode packet! java.lang.IllegalArgumentException: No provider for argument shetiphiancore:team_string
[00:10:29 INFO]: [djytw] disconnected with: The server you were previously on went down, you have been connected to a fallback server
```
(Velocity has a similar report)

**Additional context**
Minecraft Version: `1.18.2`
Forge Version:  `Fabric Loader 0.13.3`
ShetiPhian Core Version:  `shetiphiancore 1.2.5`
My Mod(s) Name & Version: `enderchests 1.0.6` `cloth-config 6.2.57`
Are other mods present? `No`

**Comment by ShetiPhian** [comment]
My thoughts on this is if direct connections work but proxy servers don't, then the proxy is the focus.
Since I'm not familiar with either, have heard of Bungee but that's it, I just ran some Google searches.


From the FabricProxy Dev
> Because BungeeCord is not compatible with Fabric-API and most mods, I decided to drop BungeeCord support.

All of my mods use the Fabric-API so BungeeCord will never work.
Turning my attention to Velocity.

FabricProxy-Lite might help
https://www.curseforge.com/minecraft/mc-mods/fabricproxy-lite

But also their FAQ has two entries that may help. (https://velocitypowered.com/wiki/users/faq/)

**Invalid Payload Register**
> [server connection] player1 -> hub has connected
> [connected player] player1 (/localhost:58943): kicked from server hub: Invalid payload REGISTER!

You can fix this issue if you use Paper (or a fork of Paper) 1.12.2 or above by adding the startup flag -`Dpaper.disableChannelLimit=true` to the server's startup flags and restarting the server.


**Argument type identifier XXX unknown**
> Argument type identifier <namespace>:<name> unknown.

If you run a modded server and are using Fabric 1.16+ and above, update to at least Velocity 1.1.2 and install [CrossStitch](https://www.curseforge.com/minecraft/mc-mods/crossstitch).

**Comment by AchiraE** [comment]
Super! I solved the problem by installing CrossStitch. Thanks for your help!

FabricProxy-Lite actually has nothing to do with this. I've already installed it before and it just used to pass UUID from proxy to the server

I also tried to look into this. The problem is that the proxies does not know how to handle the custom Brigadier argument providers, so register the provider in proxy server is needed. For Velocity, custom providers can be wrapped in crossstitch namespace, which could be resolved by Velocity; for Bungee, no way to register (or maybe writing a Bungee plugin with some dirty reflections  -- then why not use Velocity).
